kolibri/core/auth/models.py

```diff
@@ -840,8 +847,23 @@ def session_data(self):
             "kind": roles,
             "can_manage_content": self.can_manage_content,
             "facility_id": self.facility_id,
+            # Is this user a member of a facility that has been fully imported?
+            "full_facility_import": self.full_facility_import,
         }
 
+    @cached_property
+    def full_facility_import(self):
+        """
+        Returns True if this user is a member of a facility that has been fully imported.
+        """
+        return (
+            Certificate.objects.get(id=self.dataset_id)
+            .get_descendants(include_self=True)
+            .exclude(_private_key__isnull=True)
+            .filter(scope_definition_id=ScopeDefinitions.FULL_FACILITY)
+            .exists()
+        )
```

Comment:
Yes, I didn't implement any caching at first, as it's only being used within the session endpoint currently, but could definitely be cached via the `dataset_id`. Happy to add the cache for this, I can do a quick query check on the session endpoint to see how much it would save us.

Reply:
I was thinking we could use this to refine the logic for SoUD sync queuing, which currently goes through all users
https://github.com/learningequality/kolibri/blob/700f7095006b44090488e4d82632a875a38059dd/kolibri/plugins/learn/kolibri_plugin.py#L120-L123